main.py

```diff
@@ -16,17 +18,21 @@ def main():
     # XXX: Give player level boost for testing purposes
     player.level = 10
 
+    # Initialize Dungeon
+    dungeon = Dungeon()
+    dungeon.initialize(player)
+
     # Initialize Display Manager
     display_manager = DisplayManager()
-    display_manager.gen_map(player)
-    display_manager.cur_map.entities.append(player)
+    display_manager.player = player
+    display_manager.dungeon = dungeon
```

Comment:
This is a big no-no, you should not inject global variables into a module, this is called monkey-patching and should only be done as a last resource with third-party libs/programs, you can pass them as arguments to the DisplayManager constructor instead.